### PR TITLE
FSDP optimizer state channels last fix

### PR DESCRIPTION
**Pull request by pzharrington**
# PhysicsNeMo Pull Request

## Description

Fixes a silent corruption of optimizer state when checkpointing FSDP-wrapped models that have `channels_last` params (e.g. DiT in stormcast, or any SongUNet trained with apex GroupNorm).

PyTorch FSDP with `use_orig_params=False` packs and unpacks the `FlatParameter` asymmetrically for non-truly-contiguous params:

- **Save** (`_get_unflat_views` in `torch/distributed/fsdp/_flat_param.py`)
  uses `as_strided((numel,),(1,))` -- reads bytes in **storage** order.
- **Load** (`_flatten_tensor_optim_state` in `_optim_utils.py`)
  uses `torch.flatten` -- reads bytes in **logical** (row-major) order.

For a 4-D Conv2d weight in `channels_last` format, those two byte orders differ. After a save/load round-trip the optimizer state for that param is silently scrambled, while the model weights survive (the model load path uses the storage-aware variant on both ends).

Fixes add a new helper to `physicsnemo/utils/checkpoint.py` that detects the case and performs appropriate permute+contiguous+view calls to load state correctly. Adds test coverage in package and recipe unit tests for the case. This PR also removes train dataloader seeding in the recipe as requested by SAs.

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: test/utils/test_checkpoint_distributed.py [hunk 1]
```diff
@@ -477,7 +478,7 @@ def _build_distributed_model():
     if dm.rank == 0:
         for key in ref_params:
             assert torch.allclose(
-                ref_params[key].cpu(), loaded_params[key].cpu(), rtol=1e-5, atol=1e-5
+                ref_params[key].cpu(), loaded_params[key].cpu(), rtol=1e-4, atol=1e-4
```
[the remark this answers is not in the document]
Comment (pzharrington) [reply test/utils/test_checkpoint_distributed.py]: Yes, needed to stop a flaky failure. 1e-5 was too tight for comparing outputs of forward pass between sharded and unsharded modes.

File: physicsnemo/utils/checkpoint.py [hunk 2]
```diff
@@ -223,6 +223,62 @@ def _redistribute_optim_sd_for_dtensor(
...
+    if "state" not in optim_sd:
+        return optim_sd
+
+    def _maybe_remap(t: torch.Tensor) -> torch.Tensor:
+        if isinstance(t, DTensor):
+            return t
+        if t.dim() == 4 and t.is_contiguous(memory_format=torch.channels_last):
+            return t.permute(0, 2, 3, 1).contiguous().view(*t.shape)
+        if t.dim() == 5 and t.is_contiguous(memory_format=torch.channels_last_3d):
+            return t.permute(0, 2, 3, 4, 1).contiguous().view(*t.shape)
```
[the remark this answers is not in the document]
Comment (pzharrington) [reply physicsnemo/utils/checkpoint.py:289]: Thanks, fixed

**Review by CharlelieLrt — APPROVED** [review]
LGTM!
